datasync/chaindatafetcher/kas/repository_test.go

```diff
@@ -131,9 +137,52 @@ func (s *SuiteRepository) SetupSuite() {
 	}
 
 	setTestDatabase(s.T(), mysql)
-	s.repo = &repository{db: mysql}
+	s.repo = &repository{db: mysql, config: DefaultKASConfig}
 }
 
 func TestSuite(t *testing.T) {
 	suite.Run(t, new(SuiteRepository))
 }
+
+func Test_repository_InvalidateCacheEOAList(t *testing.T) {
+	testXChainId := "test-xkrn"
+	testAuth := "test-auth"
+	testEOAs := make(map[common.Address]struct{})
+	numEOAs := 10
+	for i := 0; i < numEOAs; i++ {
+		testEOAs[*genRandomAddress()] = struct{}{}
+	}
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		assert.Equal(t, testXChainId, req.Header.Get("x-chain-id"))
+		assert.Equal(t, testAuth, req.Header.Get("Authorization"))
+
+		buffer := make([]byte, 1024)
+		bodyStr := ""
+
+		for {
+			n, err := req.Body.Read(buffer)
+			bodyStr += string(buffer[:n])
+			if err == io.EOF {
+				break
+			}
+		}
+
+		var body map[string]interface{}
+
+		assert.NoError(t, json.Unmarshal([]byte(bodyStr), &body))
```

Comment:
This server should check if the passed map is equal to `testEOAs`.

Reply:
Right. Applied. 59bbdf8a1a8e7149092b3d6ec298c460402d664d
Thanks for your review.